src/py/flwr/client/clientapp/clientappio_servicer.py

```diff
@@ -122,6 +125,24 @@ def PushClientAppOutputs(
             proto_status = clientappstatus_to_proto(status=status)
             return PushClientAppOutputsResponse(status=proto_status)
 
+    def GetToken(
+        self, request: GetTokenRequest, context: grpc.ServicerContext
+    ) -> GetTokenResponse:
+        """Get token."""
+        log(DEBUG, "ClientAppIo.GetToken")
+        res = GetTokenResponse()
+        if self.token:
+            # If token is set, use it in response
+            res.token = self.token
+            # Resetting token
+            self.token = None
+        return res
+
+    def set_token(self, token: int) -> None:
+        """Set the token."""
+        log(DEBUG, "ClientAppIo.set_token")
+        self.token = token
```

Comment:
Related to the comment above, this can probably be removed since the token gets set implicitly via `set_inputs`.